### PR TITLE
[DRAFT] OpenTelemetry documentation and yamls

### PR DESCRIPTION
**Pull request by SchneiderMalte**
openTelemetry integration with Quarkus version 1.13.0Final. DISCLAIMER: Due to incompatibility with tkit-quarkus-jpa based on Quarkus version 1.12 currently not executable.

### CODE REVIEW

File: reference-project/prometheus.yaml [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
     metrics_path: "/q/metrics"
     static_configs:
-      - targets: ["host.docker.internal:8080"]
\ No newline at end of file
+      - targets: ["host.docker.internal:8080"]
+      #- targets: ['otel-collector:8889']
+      #- targets: ['otel-collector:8888']
+
+#remote_write:
+ # - url: http://victoriametrics:8428/api/v1/write
```
Comment (maybeec) [inline reference-project/prometheus.yaml:12]: ifi not needed, please remove the comments

File: reference-project/src/main/java/com/devonfw/application/quarkus/sample/animalmanagement/logic/impl/AnimalDAO.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import javax.persistence.criteria.Predicate;
 import javax.persistence.criteria.Root;
 
-import org.eclipse.microprofile.opentracing.Traced;
+//import org.eclipse.microprofile.opentracing.Traced;
```
Comment (maybeec) [inline reference-project/src/main/java/com/devonfw/application/quarkus/sample/animalmanagement/logic/impl/AnimalDAO.java:16]: same here.

File: reference-project/src/main/java/com/devonfw/application/quarkus/sample/animalmanagement/logic/impl/AnimalDAO.java [hunk 3]
```diff
@@ -32,7 +33,7 @@
 @ApplicationScoped
 @Slf4j
 // add all operations on this bean to tracer
-@Traced
+//@Traced
```
Comment (maybeec) [inline reference-project/src/main/java/com/devonfw/application/quarkus/sample/animalmanagement/logic/impl/AnimalDAO.java:36]: same

File: reference-project/src/main/java/com/devonfw/application/quarkus/sample/animalmanagement/logic/impl/AnimalDAO.java [hunk 4]
```diff
@@ -66,12 +68,13 @@
       cq.where(predicates.toArray(new Predicate[0]));
     }
     // we can add useful info to our span
-    this.tracer.activeSpan().setTag("predicates", predicates.size()).setTag("layer", "dao");
+    //this.tracer.activeSpan().setTag("predicates", predicates.size()).setTag("layer", "dao");
```
Comment (maybeec) [inline reference-project/src/main/java/com/devonfw/application/quarkus/sample/animalmanagement/logic/impl/AnimalDAO.java:71]: why is this uncommented?

File: reference-project/src/main/resources/application.properties [hunk 5]
```diff
@@ -16,14 +16,14 @@ quarkus.tkit.log.ignore.pattern=.*MapperImpl
...
-quarkus.jaeger.sampler-type=const
+#quarkus.jaeger.sampler-type=const
 #1=trace every single method
-quarkus.jaeger.sampler-param=1
-quarkus.log.console.format=%d{HH:mm:ss} %-5p traceId=%X{traceId}, parentId=%X{parentId}, spanId=%X{spanId}, sampled=%X{sampled} [%c{2.}] (%t) %s%e%n
+#quarkus.jaeger.sampler-param=1
+#quarkus.log.console.format=%d{HH:mm:ss} %-5p traceId=%X{traceId}, parentId=%X{parentId}, spanId=%X{spanId}, sampled=%X{sampled} [%c{2.}] (%t) %s%e%n
 #report spans to central aggregator(you must start Jaeger yourself - e.g. using the docker-compose.yaml)
-quarkus.jaeger.endpoint=http://localhost:14268/api/traces
+#quarkus.jaeger.endpoint=http://localhost:14268/api/traces
```
Comment (maybeec) [inline reference-project/src/main/resources/application.properties:26]: all not needed?

**Review by maybeec — CHANGES_REQUESTED** [review]

**Review by hohwille — COMMENTED** [review]
@SchneiderMalte thanks for this PR and your great work 👍 
This looks like a good improvement to me.
If only have the following questions:
* there seems to be no CI established for this repo, therefore I can only give a rought gut feeling about the quality of this change. Would it make sense that you give a demo in one of our standups so we ensure it is working (I believe you have tested this but working without CI gives me a bad feeling and we should also create another story to add github actions)
* You so far addressed REST. This is the most important aspect so far and great for the first PR so I am happy to merge this. However, to complete this story on the long run we should not forget about other aspects. What about opentelemtry for Kafka, JMS, JDBC, etc.? Please do not get me wrong: I do not want to have this all in this PR. We only need to add these aspects to our backlog before closing the opentelemetry story.

**Comment by hohwille** [comment]
BTW: I changed the title to give a chance what this PR is about. You added the `[DRAFT]` prefix.
Can you follow github features for this next time?
https://github.blog/2019-02-14-introducing-draft-pull-requests/
Thanks

**Pull request merged**